projects/vdk-control-cli/src/vdk/internal/control/command_groups/login_group/login.py

```diff
@@ -112,17 +110,28 @@ def login(
     oauth2_exchange_url,
 ):
     if auth_type == LoginTypes.CREDENTIALS.value:
-        if not client_id or not oauth2_discovery_url or not oauth2_exchange_url:
-            click.echo(
-                f"Login type: {auth_type} requires arguments:, --client-secret, --client-id, "
-                f"--oauth2-exchange-url, --oauth2-discovery-url"
+        try:
+            auth = Authentication(
+                client_id=client_id,
+                client_secret=client_secret,
+                auth_discovery_url=oauth2_discovery_url,
+                authorization_url=oauth2_exchange_url,
+                auth_type=auth_type,
+                cache_locally=True,
             )
-        else:
-            redirect_auth = RedirectAuthentication(
-                client_id, client_secret, oauth2_discovery_url, oauth2_exchange_url
-            )
-            redirect_auth.authentication_process()
+            auth.authenticate()
             click.echo("Login Successful")
+        except VDKAuthException as e:
+            if (
```

Comment:
Parsing error messages  is not a good practice. You should not look at what we do in Impala as a good example. We have no control over what Expcetion Impala raises so we have to parse them there. We do within vdk-control-api-auth library. We should use error codes or enums or exception hierarchy to handle this .